pkg/controller/worker/machines.go

```diff
@@ -35,17 +36,29 @@ import (
 	machinev1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
 	"github.com/pkg/errors"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
-// MachineClassKind yields the name of the AWS machine class.
+// MachineClassKind yields the name of the machine class kind used by AWS provider.
 func (w *workerDelegate) MachineClassKind() string {
-	return "AWSMachineClass"
+	return "MachineClass"
 }
 
-// MachineClassList yields a newly initialized AWSMachineClassList object.
+// MachineClassList yields a newly initialized MachineClassList object.
 func (w *workerDelegate) MachineClassList() runtime.Object {
-	return &machinev1alpha1.AWSMachineClassList{}
+	return &machinev1alpha1.MachineClassList{}
+}
+
+// cleanupOldMachineClasses sets deletionTimestamp on any older version machine class CRs.
+func (w *workerDelegate) cleanupOldMachineClasses(ctx context.Context, namespace string, machineClassList runtime.Object, wantedMachineDeployments worker.MachineDeployments) error {
+	if err := w.Client().List(ctx, machineClassList, client.InNamespace(namespace)); err != nil {
+		return err
+	}
+
+	return meta.EachListItem(machineClassList, func(machineClass runtime.Object) error {
+		return w.Client().Delete(ctx, machineClass)
+	})
```

Comment:
What about using `w.Client().DeleteAllOf()` ?

Reply:
Yes, sure. Makes sense. Will try it out.

Reply:
I have addressed this concern here - https://github.com/gardener/gardener-extension-provider-aws/pull/212/commits/f49729fe9216e86c8356953614881aac144e4c20. Nice suggestion. Thanks :)